parser/prism/Translator.cc

```diff
@@ -926,13 +880,7 @@ std::unique_ptr<parser::Node> Translator::translateStatements(pm_statements_node
     }
 
     // For multiple statements, convert each statement and add them to the body of a Begin node
-    parser::NodeVec sorbetStmts;
-    sorbetStmts.reserve(prismStmts.size());
-
-    for (auto &statement : prismStmts) {
-        unique_ptr<parser::Node> sorbetStmt = translate(statement);
-        sorbetStmts.emplace_back(std::move(sorbetStmt));
-    }
+    parser::NodeVec sorbetStmts = translateMulti(prismStmts);
```

Comment:
Do you think it would be possible to refactor this so we don't make a span in this method and instead delegate that to the `translateMulti` method?

Reply:
I played with this, but had mixed feelings. WDYT:

```diff
-   auto prismStmts = absl::MakeSpan(stmtsNode->body.nodes, stmtsNode->body.size);
    // For a single statement, do not create a `Begin` node and just return the statement, if that's enabled.
-   if (inlineIfSingle && prismStmts.size() == 1) {
+   if (inlineIfSingle && stmtsNode->body.size == 1) {
-       return translate(stmtsNode->body.nodes[0]);
    }

    // For multiple statements, convert each statement and add them to the body of a Begin node
-   parser::NodeVec sorbetStmts = translateMulti(prismStmts);
+   parser::NodeVec sorbetStmts = translateMulti(stmtsNode->body);
```

Reply:
Turns out this was the only call site that used the span overload. 🔥